temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -1137,4 +1137,20 @@ public void sleep(Duration duration) {
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Temporal server times out long poll calls after 1 minute and returns an empty result. After
+   * which the request has to be retried by the client if it wants to continue waiting. We emulate
+   * this behavior here.
+   *
+   * @return minimum between the context deadline and maximum long poll deadline.
+   */
+  private Deadline getLongPollDeadline() {
+    Deadline deadline = Context.current().getDeadline();
+    Deadline maximumDeadline =
+        Deadline.after(
+            WorkflowServiceStubsOptions.DEFAULT_SERVER_LONG_POLL_RPC_TIMEOUT.toMillis(),
+            TimeUnit.MILLISECONDS);
+    return deadline != null ? deadline.minimum(maximumDeadline) : maximumDeadline;
```

Comment:
Might be nice to have a way to override this so the test doesn't have to be so long to exercise it, but really not a big deal. Up to you if you think it's more annoying than it's worth.